File: python-pr/52/python/code-quality-readability-maintainability/batch_processor_23.py
# {fact rule=code-quality-readability-maintainability@v1.0 defects=1}
def process_batch(items, batch_size=10):
    results = []
    for i in range(0, len(items), batch_size):
        batch = items[i:i+batch_size]
        batch_result = []
        for item in batch:
            try:
                processed = process_item(item)
                batch_result.append(processed)
            except Exception as e:
                batch_result.append({'error': str(e)})
        results.extend(batch_result)
    return results
# {/fact}

File: js-pr/56/javascript/code-quality-readability-maintainability/batch_processor_23.js
// {fact rule=code-quality-readability-maintainability@v1.0 defects=1}
function processBatch(items, batchSize = 10) {
    let results = [];
    for (let i = 0; i < items.length; i += batchSize) {
        let batch = items.slice(i, i + batchSize);
        let batchResult = [];
        for (let item of batch) {
            try {
                let processed = processItem(item);
                batchResult.push(processed);
            } catch (e) {
                batchResult.push({error: e.message});
            }
        }
        results.push(...batchResult);
    }
    return results;
}
// {/fact}